mlp.py

```diff
@@ -0,0 +1,130 @@
+from sklearn.metrics import accuracy_score
+from sklearn.model_selection import train_test_split
+from sklearn.preprocessing import OneHotEncoder
+import numpy as np
+from utils import *
+
+
+class MultiLayerPerceptron:
+    def __init__(self, layers):
+        self.layers = layers
+
+        self.params = {}
+
+    def initialize_parameters(self):
+        for l in range(1, len(self.layers)):
+            in_size = self.layers[l - 1]
+            out_size = self.layers[l]
+
+            self.params["w" + str(l)] = np.random.randn(out_size, in_size) / np.sqrt(
+                in_size
+            )
+            self.params["b" + str(l)] = np.zeros((out_size, 1))
+
+    def model_forward(self, X):
+        number_of_layers = len(self.layers) - 1
+        a = X
+        caches = {}
+
+        for l in range(1, number_of_layers):
+            a_prev = a
+            weight_string = "w" + str(l)
+            bias_string = "b" + str(l)
+            w = self.params[weight_string]
+            b = self.params[bias_string]
+            a, cache = linear_forward_activation(a_prev, w, b, "relu")
+            caches[str(l)] = cache
+
+        weight_string = "w" + str(number_of_layers)
+        bias_string = "b" + str(number_of_layers)
+
+        w = self.params[weight_string]
+        b = self.params[bias_string]
+
+        al, cache = linear_forward_activation(a, w, b, "sigmoid")
+        caches[str(number_of_layers)] = cache
+
+        return (al, caches)
+
+    def model_backward(self, al, y, caches):
+        grads = {}
+        num_layers = len(self.layers) - 1
+        dal = -(y / al - (1 - y) / (1 - al))
+
+        current_cache = caches[str(num_layers)]
+        da_prev, dw, db = linear_backward_activation(dal, current_cache, "sigmoid")
+
+        weight_string = "dw" + str(num_layers)
+        bias_string = "db" + str(num_layers)
+        activation_string = "dA" + str(num_layers)
```

Comment:
**Suggestion:** Use f-string formatting for generating weight and bias strings in the model_backward function. This will make the code more readable and concise.
```suggestion
        weight_string = f"dw{num_layers}"
        bias_string = f"db{num_layers}"
        activation_string = f"dA{num_layers}"
```